include/SugarObjects/templates/basic/Basic.php

```diff
@@ -134,24 +179,23 @@ public function getEmailAddressConfirmOptIn($emailField)
      * @throws RuntimeException
      * @throws InvalidArgumentException
      */
-    private function getEmailAddressId($emailField)
+    protected function getEmailAddressId($emailField)
     {
-        $this->getEmailAddressValidateArguments($emailField);
+        global $log;
 
         $emailAddress = $this->cleanUpEmailAddress($this->{$emailField});
 
         if (!$emailAddress) {
-            global $log;
```

Comment:
Load and use the `$log` only if it's needed.

Reply:
So $log is used twice in this method, it two separate scopes. So we need to move the global out of this if statement (or scope) to guarantee that it can be accessed in both cases. see line 208.

Reply:
ok